Jenkinsfile

```diff
@@ -0,0 +1,124 @@
+			steps{
+				script{
+					sh "wget http://current.geneontology.org/ontology/go.obo"
+					sh "wget http://current.geneontology.org/ontology/external2go/ec2go"
+					sh "mv go.obo src/main/resources/"
+					sh "mv ec2go src/main/resources/"
+				}
+			}
+		}
+		// This stage builds the jar file using maven.
+		stage('Setup: Build jar file'){
+			steps{
+				script{
+					sh "mvn clean compile assembly:single"
+				}
+			}
+		}
+		// This stage executes the GOUpdate jar file. 
+		stage('Main: GO Update'){
+			steps {
+				script{
+					withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
+						sh "java -Xmx${env.JAVA_MEM_MAX}m -jar target/go-update-*-jar-with-dependencies.jar $ConfigFile"
+					}
+				}
+			}
+		}
+		// This stage backs up the gk_central and slice_current databases after they have been modified.
+		stage('Post: Backup gk_central after modifications'){
+			steps{
+				script{
+					withCredentials([usernamePassword(credentialsId: 'mySQLCuratorUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+						def central_after_update_go_update_dump = "${env.GK_CENTRAL}_${currentRelease}_after_go_update.dump"
+						sh "mysqldump -u$user -p$pass -h${env.CURATOR_SERVER} ${env.GK_CENTRAL} > $central_after_update_go_update_dump"
+						sh "gzip -f $central_after_update_go_update_dump"
+					}
+				}
+			}
+		}
+		// This stage archives the contents of the 'reports' folder generated by GO Update and sends them in an email to the default recipients list.
+		stage('Post: Email GO Update Reports'){
+			steps{
+				script{
+					sh "tar zcf go-update-v${currentRelease}-reports.tgz reports/"
+					emailext (
+						body: "Hello,\n\nThis is an automated message from Jenkins regarding an update for v${currentRelease}. The GO Update step has completed. Please review the reports attached to this email. If they look correct, these reports need to be uploaded to the Reactome Drive at Reactome>Release>Release QA>V${currentRelease}_QA>V${currentRelease}_QA_GO_Update_Reports. If they don't look correct, please email the developer running Release. \n\nThanks!",
+						to: '$DEFAULT_RECIPIENTS',
+						from: "${env.JENKINS_RELEASE_EMAIL}",
+						subject: "GO Update Reports for v${currentRelease}",
+						attachmentsPattern: "**/go-update-v${currentRelease}-reports.tgz"
+					)
+				}
+			}
+		}
+		// All databases, logs, and data files generated by this step are compressed before moving them to the Reactome S3 bucket. 
+		// All files are then deleted.
+		stage('Post: Archive Outputs'){
+			steps{
+				script{
+					def s3Path = "${env.S3_RELEASE_DIRECTORY_URL}/${currentRelease}/go_update"
+					sh "mkdir -p databases/"
+					sh "mkdir -p data/"
+					sh "mv --backup=numbered *_${currentRelease}_*.dump.gz databases/"
+					sh "mv src/main/resources/go.obo data/"
+					sh "mv src/main/resources/ec2go data/"
+					sh "gzip data/*"
+					sh "mv go-update-v${currentRelease}-reports.tgz data/"
+					sh "gzip logs/*"
```

Comment:
I'm not sure, but the gzip might generate errors if there is already gzipped content in the directories.  Maybe it could be `gzip !(*.gz)`

Reply:
When I do a test run of this, it has a message that says 'thing.gz already has a .gz suffix' -- More informational than error

Reply:
I think that message might be written to STDERR.  I'm not sure if Jenkins watches for that to see if something went wrong?  I'm okay with leaving it just as `gzip *`.  I just thought I'd mention that it will output that message if there is a file that already is gzipped.

Reply:
OK I'll do a test run of this before merging

Reply:
Yea after testing this, it doesn't cause the Jenkins build terminate, just outputs that message.